src/py/mat3ra/made/tools/build/defect/builders.py

```diff
@@ -410,27 +414,23 @@ def create_island(
         Returns:
             The material with the island added.
         """
-
         new_material = material.clone()
         original_max_z = get_atomic_coordinates_extremum(new_material, use_cartesian_coordinates=False)
         material_with_additional_layers = self.create_material_with_additional_layers(new_material, thickness)
         added_layers_max_z = get_atomic_coordinates_extremum(material_with_additional_layers)
-
         if condition is None:
+            condition = self._default_condition
 
-            def condition(coordinate: List[float]):
-                return True
-
+        thickness_nudge_value = (added_layers_max_z - original_max_z) / thickness
```

Comment:
what is this `thickness_nudge_value`?? Is there a better name? If not - explain in a comment

Reply:
`thickness_nudge_value_z` + comment

Reply:
I'll switch to cartesians and do this the proper way. Wasn't clear at the time